lightyear/src/transport/websockets/client_native.rs

```diff
@@ -0,0 +1,242 @@
+                                    code.unwrap_or(1000),
+                                    reason.unwrap_or("".to_string()).as_bytes(),
+                                ))
+                                .await;
+                            break;
+                        }
+                        WebSocketMessage::Binary(data) => {
+                            trace!("sending packet to server!: {:?}", &data);
+                            send.lock()
+                                .await
+                                .write_frame(Frame::binary(Payload::Owned(data)))
+                                .await;
+                        }
+                    }
+                }
+            }
+        });
+    }
+
+    fn get_tls_connector(&self) -> TlsConnector {
+        let root_store = RootCertStore {
+            roots: webpki_roots::TLS_SERVER_ROOTS.to_vec()
+        };
+    
+        let config = ClientConfig::builder()
+          .with_root_certificates(root_store)
+          .with_no_client_auth();
+      
+        TlsConnector::from(Arc::new(config))
+    }
+}
+
+impl Transport for WebSocketClientSocket {
+    fn local_addr(&self) -> SocketAddr {
+        self.server_addr
+    }
+
+    fn listen(self) -> (Box<dyn PacketSender>, Box<dyn PacketReceiver>) {
+        let (to_server_sender, to_server_receiver) = mpsc::unbounded_channel();
+        let (from_server_sender, from_server_receiver) = mpsc::unbounded_channel();
+
+        let packet_sender = WebSocketClientSocketSender { to_server_sender };
+
+        let packet_receiver = WebSocketClientSocketReceiver {
+            buffer: [0; MTU],
+            server_addr: self.server_addr,
+            from_server_receiver,
+        };
+
+        tokio::spawn(async move {
+            info!("Starting server websocket task");
+
+            let tcp_stream = TcpStream::connect(self.server_addr).await.unwrap();
+
+            let (protocol, host) = if let Some(tls_config) = &self.tls_config {
+                ("wss", tls_config.server_name)
+            } else {
+                ("ws", format!("{}:{}", self.server_addr.ip(), self.server_addr.port()))
+            };
+
+            let req = Request::builder()
+                .method("GET")
+                .uri(format!("{}://{}:{}/", protocol, self.server_addr.ip(), self.server_addr.port()))
+                .header(HOST, host)
+                .header(UPGRADE, "websocket")
+                .header(CONNECTION, "upgrade")
+                .header(SEC_WEBSOCKET_KEY, generate_key())
+                .header(SEC_WEBSOCKET_VERSION, "13")
+                .body(Empty::<Bytes>::new())
+                .unwrap();
+
+            let ws = if let Some(tls_config) = &self.tls_config {
+                let tls_stream = self.get_tls_connector().connect(ServerName::try_from(tls_config.server_name).unwrap(), tcp_stream).await.unwrap();
+                client(&SpawnExecutor, req, tls_stream).await.unwrap().0
```

Comment:
I wouldn't use the name `client` directly here as it's a bit ambiguous, maybe rename as `wsclient`?